### ISSUE TITLE
Issue with extracting features. 

### ISSUE THREAD

**Issue by DavidGill159** [YttriLab/A-SOID#60]
**Description:**
The histogram produced during feature extraction seems to look quite different from the example provided on this GitHub. I am using Boris to label the syllables (1000+ manual labels across 3x 6-minute videos) and DLC to track landmarks and pose. What could be going wrong here? I am also confused by the bin size selection; is this the number of behavioural syllables to look at within each bin? What would you recommend as a good starting point? Additionally, I also have 3D (xyz) data I want to eventually use. is this iteration of a-soid able to process such?; The paper mentions you handled some 3D data. Also, regarding the input video data, should it all be from 1 cameras perspective?

![thumbnail_image](https://github.com/YttriLab/A-SOID/assets/116177739/93e81272-0c29-42da-89e0-516e48f2b037)

**Comment by JensBlack** [comment]
That is clearly an unexpexted output. Have you changed something in the code?

if not, it would be great if you could send me the package versions that where installed in your repository. Specifically from `plotly`

**Comment by DavidGill159** [comment]
@JensBlack Hi, Nope, i have not changed the code. Here is a list of all the packages in my asoid environment:

# Name                    Version                   Build  Channel
altair                    5.2.0                    pypi_0    pypi
asoid                     0.3                       dev_0    <develop>
asttokens                 2.4.1                    pypi_0    pypi
attrs                     23.1.0                   pypi_0    pypi
backcall                  0.2.0                    pypi_0    pypi
backports-zoneinfo        0.2.1                    pypi_0    pypi
beautifulsoup4            4.12.2                   pypi_0    pypi
blinker                   1.7.0                    pypi_0    pypi
bokeh                     3.1.1                    pypi_0    pypi
bs4                       0.0.1                    pypi_0    pypi
ca-certificates           2023.08.22           haa95532_0
cachetools                5.3.2                    pypi_0    pypi
certifi                   2023.11.17               pypi_0    pypi
charset-normalizer        3.3.2                    pypi_0    pypi
click                     8.1.7                    pypi_0    pypi
colorama                  0.4.6                    pypi_0    pypi
compress-pickle           2.1.0                    pypi_0    pypi
contourpy                 1.1.1                    pypi_0    pypi
cycler                    0.12.1                   pypi_0    pypi
cython                    0.29.36                  pypi_0    pypi
decorator                 4.4.2                    pypi_0    pypi
executing                 2.0.1                    pypi_0    pypi
ffmpeg                    4.2.2                he774522_0
ffmpeg-python             0.2.0                    pypi_0    pypi
fonttools                 4.46.0                   pypi_0    pypi
future                    0.18.3                   pypi_0    pypi
gitdb                     4.0.11                   pypi_0    pypi
gitpython                 3.1.40                   pypi_0    pypi
h5py                      3.7.0                    pypi_0    pypi
hdbscan                   0.8.33                   pypi_0    pypi
hydralit                  1.0.14                   pypi_0    pypi
hydralit-components       1.0.10                   pypi_0    pypi
idna                      3.6                      pypi_0    pypi
imageio                   2.33.0                   pypi_0    pypi
imageio-ffmpeg            0.4.9                    pypi_0    pypi
importlib-metadata        6.11.0                   pypi_0    pypi
importlib-resources       6.1.1                    pypi_0    pypi
ipython                   8.4.0                    pypi_0    pypi
jedi                      0.19.1                   pypi_0    pypi
jinja2                    3.1.2                    pypi_0    pypi
joblib                    1.3.2                    pypi_0    pypi
jsonschema                4.20.0                   pypi_0    pypi
jsonschema-specifications 2023.11.2                pypi_0    pypi
kiwisolver                1.4.5                    pypi_0    pypi
libffi                    3.4.4                hd77b12b_0
llvmlite                  0.35.0                   pypi_0    pypi
lxml                      4.9.3                    pypi_0    pypi
markdown-it-py            3.0.0                    pypi_0    pypi
markupsafe                2.1.3                    pypi_0    pypi
matplotlib                3.5.3                    pypi_0    pypi
matplotlib-inline         0.1.6                    pypi_0    pypi
mdurl                     0.1.2                    pypi_0    pypi
moviepy                   1.0.3                    pypi_0    pypi
numba                     0.52.0                   pypi_0    pypi
numpy                     1.22.4                   pypi_0    pypi
opencv-python             4.8.1.78                 pypi_0    pypi
openssl                   3.0.12               h2bbff1b_0
packaging                 23.2                     pypi_0    pypi
pandas                    1.5.3                    pypi_0    pypi
parso                     0.8.3                    pypi_0    pypi
pickleshare               0.7.5                    pypi_0    pypi
pillow                    9.2.0                    pypi_0    pypi
pip                       23.3.1           py38haa95532_0
pkgutil-resolve-name      1.3.10                   pypi_0    pypi
plotly                    5.10.0                   pypi_0    pypi
proglog                   0.1.10                   pypi_0    pypi
prompt-toolkit            3.0.41                   pypi_0    pypi
protobuf                  4.25.1                   pypi_0    pypi
psutil                    5.9.6                    pypi_0    pypi
pure-eval                 0.2.2                    pypi_0    pypi
pyarrow                   14.0.1                   pypi_0    pypi
pydeck                    0.8.1b0                  pypi_0    pypi
pygments                  2.17.2                   pypi_0    pypi
pynndescent               0.5.11                   pypi_0    pypi
pyparsing                 3.1.1                    pypi_0    pypi
python                    3.8.18               h1aa4202_0
python-dateutil           2.8.2                    pypi_0    pypi
pytz                      2023.3.post1             pypi_0    pypi
pyyaml                    6.0.1                    pypi_0    pypi
referencing               0.31.1                   pypi_0    pypi
requests                  2.31.0                   pypi_0    pypi
rich                      13.7.0                   pypi_0    pypi
rpds-py                   0.13.2                   pypi_0    pypi
scikit-learn              1.2.2                    pypi_0    pypi
scipy                     1.9.3                    pypi_0    pypi
seaborn                   0.12.2                   pypi_0    pypi
setuptools                59.8.0                   pypi_0    pypi
six                       1.16.0                   pypi_0    pypi
smmap                     5.0.1                    pypi_0    pypi
soupsieve                 2.5                      pypi_0    pypi
sqlite                    3.41.2               h2bbff1b_0
stack-data                0.6.3                    pypi_0    pypi
stqdm                     0.0.5                    pypi_0    pypi
streamlit                 1.29.0                   pypi_0    pypi
streamlit-option-menu     0.3.6                    pypi_0    pypi
tenacity                  8.2.3                    pypi_0    pypi
threadpoolctl             3.2.0                    pypi_0    pypi
toml                      0.10.2                   pypi_0    pypi
toolz                     0.12.0                   pypi_0    pypi
tornado                   6.4                      pypi_0    pypi
tqdm                      4.64.1                   pypi_0    pypi
traitlets                 5.14.0                   pypi_0    pypi
typing-extensions         4.8.0                    pypi_0    pypi
tzdata                    2023.3                   pypi_0    pypi
tzlocal                   5.2                      pypi_0    pypi
umap-learn                0.5.5                    pypi_0    pypi
urllib3                   2.1.0                    pypi_0    pypi
validators                0.22.0                   pypi_0    pypi
vc                        14.2                 h21ff451_1
vs2015_runtime            14.27.29016          h5e58377_2
watchdog                  3.0.0                    pypi_0    pypi
wcwidth                   0.2.12                   pypi_0    pypi
wheel                     0.41.2           py38haa95532_0
xyzservices               2023.10.1                pypi_0    pypi
zipp                      3.17.0                   pypi_0    pypi

**Comment by JensBlack** [comment]
> The histogram produced during feature extraction seems to look quite different from the example provided on this GitHub. I am using Boris to label the syllables (1000+ manual labels across 3x 6-minute videos) and DLC to track landmarks and pose. What could be going wrong here? 

Okay, I think I was able to recreate your use case.

The plot is using a stacked/relative represantive [Link](https://plotly.github.io/plotly.py-docs/generated/plotly.express.histogram.html) to generate this figure. For your specific case and in combination with the number of bins set, you have some bins that do not contain any counts, therefore it is more obvious that we are using a stacked version to show the distribution of bout lengths.

Here is an example of one of our own testcases with different number of bins:

*low number of bins*
![grafik](https://github.com/YttriLab/A-SOID/assets/44863941/994f0c80-a8c6-4501-b62f-7b42a5bd1097)

*higher number of bins*
![grafik](https://github.com/YttriLab/A-SOID/assets/44863941/20f5ed38-5cf7-4d58-bed4-e4f30310b98a)

*this is a version with "overlay" instead of "relative":*
![grafik](https://github.com/YttriLab/A-SOID/assets/44863941/f932fb18-4dff-40d0-9f0c-85e6e518b3a3)



The plot parameter "number of bins" is only used to help visualize the distribution of the data. The minimum bout duration is what decides the final feature extraction.

>I am also confused by the bin size selection; is this the number of behavioural syllables to look at within each bin? What would you recommend as a good starting point?

 Here I'd refer you to our latest BioRxiv version, if you are interested in learning more about how different bin sizes affect performance for the CalMS21 data set.
**TL;DR: You want a minimum duration small enough to capture you short bouts but big enough to include temporal dynamics between frames. A good start is something in the range of 100 - 400 ms.**

**Comment by DavidGill159** [comment]
@JensBlack thanks for all that useful information! only follow up I would have is my query about 3D data? I have structured it in an identical format to the DLC csvs except with the added z coordinates.

**Comment by JensBlack** [comment]
Hi David,

the 3D capabilities are currently being integrated into the app and will be available in the next update. If you are interested in testing them beforehand, I can let you know once there the branch is uploaded for testing. It would be best if you create a seperate "issue"/feature request for this, so that we can gather all info there.
